Makefile

```diff
@@ -56,6 +56,11 @@ KIND := $(TOOLS_BIN_DIR)/kind
 KUSTOMIZE := $(TOOLS_BIN_DIR)/kustomize
 TOOLING_BINARIES := $(CONTROLLER_GEN) $(CONVERSION_GEN) $(GINKGO) $(GOLANGCI_LINT) $(GOVC) $(KIND) $(KUSTOMIZE)
 
+# Set --output-base for conversion-gen if we are not within GOPATH
+ifneq ($(abspath $(ROOT_DIR)),$(shell go env GOPATH)/src/sigs.k8s.io/cluster-api-provider-vsphere)
```

Comment:
can we check If this approach is consistent across cluster-api + other providers ?

Reply:
Yeah, good question @yastij. This is currently being [implemented in cluster-api](https://github.com/kubernetes-sigs/cluster-api/pull/4349), so we may want to wait until that lands just to see if anything needs to change there.

But this is based on an existing approach from CAPZ here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/03a76e30059c9d467ee470600d3471f8bf6ba4fa

It is also handled in k/k, but in that case there is a small script (with a very large library of functions hidden behind it) that handles this: https://github.com/kubernetes/kubernetes/blob/master/hack/run-in-gopath.sh

Ultimately, I think conversion-gen code should be updated to handle running outside of GOPATH. It was designed when that was a safe assumption to make, but now it needs to be accounted for everywhere it is being used.

Reply:
PR has now merged to cluster-api, so this would match that and CAPZ. Now that cluster-api has merged, I plan on addressing the same issue in the other upstream cluster-api-provider-* repos.